pkg/k8s/init.go

```diff
@@ -271,7 +271,10 @@ func WaitForNodeInformation(ctx context.Context, k8sGetter k8sGetter) error {
 		node.SetLabels(n.Labels)
 
 		node.SetK8sExternalIPv4(n.GetExternalIP(false))
-		node.SetK8sExternalIPv6(n.GetExternalIP(true))
+		// Avoid removing the IPv4 if there is no IPv6.
+		if extIP6 := n.GetExternalIP(true); extIP6 != nil {
+			node.SetK8sExternalIPv6(extIP6)
+		}
```

Comment:
Sorry for not getting to this earlier.

Looking at the v1.12 code, I think we would have actually been fine - there `SetK8sExternalIPv6()` merely updates `addrs.ipv6ExternalAddress`, without any further side effects. But this change shouldn't cause any harm either, so no need to back it out again.